src/agent/debugger/src/stack.rs

```diff
@@ -120,18 +81,19 @@ impl Display for DebugStackFrame {
         match self {
             DebugStackFrame::Frame {
                 module_name,
-                location,
+                module_offset,
                 symbol,
+                symbol_location,
             } => {
                 if let Some(symbol) = symbol {
                     write!(formatter, "{}!{}", module_name, symbol)?;
                 } else {
                     write!(formatter, "{}", module_name)?;
                 }
-                if let Some(file_info) = &location.file_info {
-                    write!(formatter, " {}", file_info)
+                if let Some(symbol_location) = &symbol_location {
+                    write!(formatter, " {}", symbol_location)
                 } else {
-                    write!(formatter, "+0x{:x}", location.displacement)
+                    write!(formatter, "+0x{:x}", module_offset)
```

Comment:
This is looking good, I think we just lost one thing.

Before this PR, we overloaded the `DebugFunctionLocation.displacement` field, so it would either mean "offset, relative to the image base of the module" or "offset from the base of the _symbol's_ mapped image base". With this most recent change, it always means "module-relative offset" (unless I misread something). 

Suppose:
- `mymodule` is mapped to `0x1000`
- `mymodule!myfunc` is mapped to `0x1200`
- The PC is `0x1234`

When we don't have source line info, we want to see any of the following, in order of decreasing preference:
1. `mymodule!myfunc+0x34`
2.  `mymodule+0x234`  (if no symbols)
3. `<UnknownModule>+0x234`  (if our debugger APIs fail to give us the module name for some reason)
4. `<corrupt stack frame(s)>` (if we can't even use the above fallback where try to guess a module base)

As written, I think in case (1), we'd emit `mymodule!myfunc+0x234`. We really want to grab, save, and use the [`SymInfo.displacement`](https://github.com/microsoft/onefuzz/blob/main/src/agent/debugger/src/stack.rs#L197) when we can get it (under the hood, from [`SymFromInlineContextW()`](https://docs.microsoft.com/en-us/windows/win32/api/dbghelp/nf-dbghelp-symfrominlinecontextw)).

Reply:
addressed